File: P72410.py
def solution(new_id):
    spc = ["-", "_", "."]
    answer = ""
    
    for digit in new_id:
        if ord("a") <= ord(digit) <= ord('z') or ord("0") <= ord(digit) <= ord("9"):
            answer += digit
        
        elif ord("A") <= ord(digit) <= ord("Z"):
            answer += digit.lower()
        
        elif digit in spc:
            answer += digit
    
    flag = 1
    while flag:
        for i in range(len(answer)-1):
            if answer[i] == "." and answer[i] == answer[i+1]:
                answer = answer[:i] + answer[i+1:]
                break
        else:
            flag = 0
            
    if answer and answer[0] == ".":
        answer = answer[1:]
    
    if answer and answer[-1] == ".":
        answer = answer[:-1]
    
    if not answer:
        answer += "a"
    
    if len(answer) >= 16:
        answer = answer[:15]
    
    if answer and answer[-1] == ".":
        answer = answer[:-1]
    
    if answer:
        while len(answer) <= 2:
            answer += answer[-1]
            
    return answer

File: P72410.java
public class P72410 {
    public Boolean contains(char[] arr, char x) {
        for (int i=0; i<arr.length; i++) {
            if (arr[i] == x) return true;
        }
        return false;
    }
    
    public String solution(String new_id) {
        
        char[] spc = new char[] {'-', '_', '.'};
        String answer = "";
        
        for (int i=0; i<new_id.length(); i++) {
            char digit = new_id.charAt(i);
            if (('a' <= digit && digit <= 'z') || ('0' <= digit && digit <= '9')) answer += digit;
            else if ('A' <= digit && digit <= 'Z') answer += Character.toLowerCase(digit);
            else if (contains(spc, digit)) answer += digit; 
        }
        
        boolean flag = true;
        while (true) {
            flag = true;
            for (int i=0; i<answer.length()-1; i++) {
                if (answer.charAt(i) == '.' && answer.charAt(i) == answer.charAt(i+1)) {
                    answer = answer.substring(0, i) + answer.substring(i+1, answer.length());
                    flag = false;
                    break;
                }
            }
            if (flag) break;
        }
        
        if (answer.length() > 0 && answer.charAt(0) == '.') answer = answer.substring(1, answer.length());
        
        if (answer.length() > 0 && answer.charAt(answer.length()-1) == '.') answer = answer.substring(0, answer.length()-1);
        
        if (answer.length() == 0) answer += 'a';
        
        if (answer.length() >= 16) answer = answer.substring(0, 15);
        
        if (answer.length() > 0 && answer.charAt(answer.length()-1) == '.') answer = answer.substring(0, answer.length()-1);
        
        if (answer.length() > 0) {
            while (answer.length() <= 2) {
                answer += answer.charAt(answer.length()-1);
            }
        }
        
        return answer;
    }
}
